pkg/contracts/src/diamonds/Diamond.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+
+import {LibDiamond} from "./libraries/LibDiamond.sol";
+import {IDiamondCut} from "./interfaces/IDiamondCut.sol";
+import {IDiamondLoupe} from "./interfaces/IDiamondLoupe.sol";
+import {IERC173} from "./interfaces/IERC173.sol";
+//import { IERC165} from "./interfaces/IERC165.sol";
+
+import {IERC1822Proxiable} from "@openzeppelin/contracts/interfaces/draft-IERC1822.sol";
+// When no function exists for function called
+
+import {IDiamondCut} from "./interfaces/IDiamondCut.sol";
+import {LibDiamond} from "./libraries/LibDiamond.sol";
+
+
+error FunctionNotFound(bytes4 _functionSelector);
+
+// This is used in diamond constructor
+// more arguments are added to this struct
+// this avoids stack too deep errors
+struct DiamondArgs {
+    address owner;
+    address init;
+    bytes initCalldata;
+}
+
+contract Diamond is IERC1822Proxiable, IDiamondCut {
```

Comment:
Should we use this library ? https://www.npmjs.com/package/@solidstate/contracts, I think should be compatible with louper that way to add and remove function through it

Reply:
Reference: https://www.youtube.com/watch?v=8p4NhC9sLDA&list=PLlxGQpt_kNBdc_vVUF0XyVdi5c5WSWGII&index=4&sttick=0